src/components/sections/judge/JudgeContestFindingsTableRow.tsx

```diff
@@ -0,0 +1,68 @@
+import Link from 'next/link'
+import {DateTime} from 'luxon'
+import {VerifiedIcon} from 'lucide-react'
+
+import {TableCell, TableRow} from '@/components/ui/Table'
+import {ContestFinding} from '@/server/actions/finding/getFinding'
+import {formatDate} from '@/lib/utils/common/format'
+import {translateEnum} from '@/lib/utils/common/enums'
+import {Button} from '@/components/ui/Button'
+import {FindingStatus} from '@/server/db/models'
+import {PATHS} from '@/lib/utils/common/paths'
+
+type JudgeContestFindingsTableRowProps = {
+  finding: ContestFinding
+}
+
+const JudgeContestFindingsTableRow = ({
+  finding,
+}: JudgeContestFindingsTableRowProps) => {
+  const getActionButton = () => {
+    switch (finding.status) {
+      case FindingStatus.PENDING:
+        return (
+          <Button asChild variant="outline" size="small">
+            <Link href={PATHS.judgeFinding(finding.id)}>Review</Link>
+          </Button>
+        )
+      case FindingStatus.APPROVED:
+        return (
+          <Button asChild variant="outline" size="small">
+            <Link href="#">Deduplicate</Link>
+          </Button>
+        )
+      case FindingStatus.REJECTED:
+        return null
+    }
```

Comment:
Good point. I changed it so that even approved or rejected findings can be altered as long as a contest is not yet finalized. Had to change the reject logic to also remove deduplicated finding associations for the rejected finding.